### PR TITLE
Feature/pantry class

### LINKED ISSUE #23: Create Pantry Class

### LINKED ISSUE #24: Determine whether a user’s pantry has enough ingredients to cook a given meal.

### LINKED ISSUE #25: Determine the amount of missing ingredients still needed to cook a given meal, based on what’s in the user’s pantry.

### PR DESCRIPTION
**Pull request by mattruder**
- What's this PR do?
Allows a user to determine whether their pantry has enough ingredients to cook a given meal.
Allows a user to determine the amount of missing ingredients still needed to cook a given meal, based on what’s in their pantry.

- Where should the reviewer start?
scripts.js
Pantry.js
Pantry-test.js

- How should this be manually tested?
Run npm test in the main directory

- Any background context you want to provide?
N/A

- What are the relevant tickets?
Resolve #23 
Resolve #24 
Resolve #25 

- Screenshots (if appropriate)
N/A

- Questions:
None

### CODE REVIEW

File: src/classes/Pantry.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    let pantryIngredientAmount = ((this.ingredients.filter((thePantryIngredient) => {
+                      if (allRecipeIngredient.id === thePantryIngredient.ingredient) {
+                        return thePantryIngredient.amount
+                      }
+                    })))
+                    allRecipeIngredient.quantity.amount = ingredientInRecipe.quantity.amount - pantryIngredientAmount[0].amount
+                    if(!this.ingredientsNeeded.includes(allRecipeIngredient)) {
+                      this.ingredientsNeeded.push(allRecipeIngredient)
+                    }
+
```
Comment (aominhlong) [inline src/classes/Pantry.js:61]: Fix uneven spacing

File: test/Pantry-test.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                  "ingredient": 11215,
+                  "amount": 5
+                },
+                {
+                  "ingredient": 2047,
+                  "amount": 6
+                },
+                {
+                  "ingredient": 8901,
+                  "amount": 0.5
```
Comment (nhodnett) [inline test/Pantry-test.js:38]: We should double check this amount.

File: test/Pantry-test.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    it(`Should contain the user's ingredients`, () => {
+        expect(pantry.ingredients).to.deep.equal(user.pantry)
+    })
+
+    it(`Should determine whether a user's pantry has enough ingredients`, () => {
+        user.decideToCook(recipe)
+      console.log('user: ', user)
+
+// console.log('2nd user log: ', user)
```
Comment (aominhlong) [inline test/Pantry-test.js:111]: Remove any commented code

**Review by nhodnett — COMMENTED** [review]
Nice work crew, onto the fetch handling...

**Pull request merged**